packages/ui/src/views/connectors/types.ts

```diff
@@ -0,0 +1,209 @@
+
+export const GITHUB_CONNECTOR_IDENTIFIER = 'Github'
+export const TERRAFORM_CONNECTOR_IDENTIFIER = 'Terraform'
+export const AWS_KMS_CONNECTOR_IDENTIFIER = 'AWS KMS'
+
+export type ConnectorFormEntityType = {
+  type: 'connector'
+  data: {
+    type: string
+    name?: string
+    payload?: ConnectorPayloadConfig
+  }
+}
+export type HARNESS_CONNECTOR_IDENTIFIER =
+  | typeof Connectors.GITHUB
+  | typeof Connectors.TERRAFORM_CLOUD
+  | typeof Connectors.AWS_KMS
+  | typeof Connectors.DOCKER
+
+export type AnyConnectorDefinition<T = string> = {
+  type: T
+  name: string
+  category: string
+  formDefinition: IFormDefinition<InputConfigType>
+}
+
+export enum CredTypeValues {
+  ManualConfig = 'ManualConfig',
+  AssumeIAMRole = 'AssumeIAMRole',
+  AssumeRoleSTS = 'AssumeSTSRole',
+  PermanentTokenConfig = 'PermanentTokenConfig'
+}
+export enum DelegateTypes {
+  DELEGATE_OIDC = 'DelegateOidc'
+}
+// Base interfaces
+export interface ConnectorSpec {
+  auth?: AuthenticationSpec
+  delegateSelectors?: string[]
+  ignoreTestConnection?: boolean
+}
+
+export interface AuthenticationSpec {
+  type: string
+}
+
+// Base connector configuration
+export interface ConnectorPayloadConfig<T extends ConnectorSpec = ConnectorSpec> {
+  name: string
+  description?: string
+  projectIdentifier?: string
+  orgIdentifier?: string
+  identifier: string
+  tags?: string[]
+  type: string
+  spec: T
+}
+export interface ConnectorConfigDTO {
+  [key: string]: any
+}
+export interface ConnectorInfoDTO {
+  accountIdentifier?: string
+  description?: string
+  identifier: string
+  name: string
+  orgIdentifier?: string
+  parentUniqueId?: string
+  projectIdentifier?: string
+  spec: ConnectorConfigDTO
+  tags?: {
+    [key: string]: string
+  }
+  type:
+    | 'K8sCluster'
+    | 'Git'
+    | 'Splunk'
+    | 'AppDynamics'
+    | 'Prometheus'
+    | 'Dynatrace'
+    | 'Vault'
+    | 'AzureKeyVault'
+    | 'DockerRegistry'
+    | 'Local'
+    | 'AwsKms'
+    | 'GcpKms'
+    | 'AwsSecretManager'
+    | 'Gcp'
+    | 'Aws'
+    | 'Azure'
+    | 'Artifactory'
+    | 'Jira'
+    | 'Nexus'
+    | 'Github'
+    | 'Gitlab'
+    | 'Bitbucket'
+    | 'Codecommit'
+    | 'CEAws'
+    | 'CEAzure'
+    | 'GcpCloudCost'
+    | 'CEK8sCluster'
+    | 'HttpHelmRepo'
+    | 'NewRelic'
+    | 'Datadog'
+    | 'SumoLogic'
+    | 'PagerDuty'
+    | 'CustomHealth'
+    | 'ServiceNow'
+    | 'ErrorTracking'
+    | 'Pdc'
+    | 'AzureRepo'
+    | 'Jenkins'
+    | 'OciHelmRepo'
+    | 'CustomSecretManager'
+    | 'ElasticSearch'
+    | 'GcpSecretManager'
+    | 'AzureArtifacts'
+    | 'Tas'
+    | 'Spot'
+    | 'Bamboo'
+    | 'TerraformCloud'
+    | 'SignalFX'
+    | 'Harness'
+    | 'Rancher'
+    | 'JDBC'
+    | 'Zoom'
+    | 'MsTeams'
+    | 'Slack'
+    | 'Confluence'
+  uniqueId?: string
+}
+export interface ConnectorType {
+  [key: string]: ConnectorInfoDTO['type']
+}
+export const Connectors: ConnectorType = {
```

Comment:
@cjlee01 @shaurya-harness How are we planning to dedupe these types? We have the same stuff in NG UI also.

Reply:
@cjlee01 these should be passed down from NGUI